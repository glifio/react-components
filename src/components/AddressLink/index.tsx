import styled from 'styled-components'
import PropTypes from 'prop-types'
import { useMemo, useCallback } from 'react'
import truncateAddress from '../../utils/truncateAddress'
import { SmartLink } from '../Link/SmartLink'
import { Label } from '../Typography'
import { CopyText } from '../Copy'
import Box from '../Box'

const Link = styled(SmartLink)`
  color: var(--purple-medium);
  text-decoration: none;

  &:hover {
    text-decoration: underline;
  }
`

export const AddressLink = ({
  id,
  address,
  label,
  disableLink,
  stopPropagation,
  hideCopy,
  hideCopyText
}: AddressLinkProps) => {
  const truncated = useMemo(
    () => (address ? truncateAddress(address) : ''),
    [address]
  )
  const explorerUrl =
    process.env.NEXT_PUBLIC_EXPLORER_URL ||
    'https://explorer-calibration.glif.link'
  const linkText = truncated && id ? `(${id})` : id || truncated
  const linkHref = `${explorerUrl}/actor/?address=${id || address}`
  const onClick = useCallback(
    (e: MouseEvent) => stopPropagation && e.stopPropagation(),
    [stopPropagation]
  )
  return (
    <Box>
      {label && (
        <Label color='core.darkgray' fontSize={1}>
          {label}
        </Label>
      )}
      <Box display='flex' flexDirection='row' gridGap='0.25em'>
        {truncated && id && <span>{truncated}</span>}
        {disableLink ? (
          <span>{linkText}</span>
        ) : (
          <Link href={linkHref} onClick={onClick}>
            {linkText}
          </Link>
        )}
        {!hideCopy && address && (
          <CopyText
            text={address}
            hideCopyText={hideCopyText}
            color='inherit'
          />
        )}
      </Box>
    </Box>
  )
}

export interface AddressLinkProps {
  id?: string
  address?: string
  label?: string
  disableLink: boolean
  stopPropagation: boolean
  hideCopy: boolean
  hideCopyText: boolean
}

AddressLink.propTypes = {
  id: PropTypes.string,
  address: PropTypes.string,
  label: PropTypes.string,
  disableLink: PropTypes.bool,
  stopPropagation: PropTypes.bool,
  hideCopy: PropTypes.bool,
  hideCopyText: PropTypes.bool
}

AddressLink.defaultProps = {
  disableLink: false,
  stopPropagation: true,
  hideCopy: false,
  hideCopyText: true
}
