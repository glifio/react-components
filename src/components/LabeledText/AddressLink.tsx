import PropTypes from 'prop-types'
import { useMemo } from 'react'

import truncateAddress from '../../utils/truncateAddress'
import {
  LabeledLink,
  LabeledLinkProps,
  LabeledLinkPropTypes
} from './LabeledLink'

const explorerUrl =
  process.env.NEXT_PUBLIC_EXPLORER_URL ||
  'https://explorer-calibration.glif.link'

export const AddressLink = ({
  id,
  address,
  label,
  color,
  disableLink,
  hideCopy,
  hideCopyText,
  shouldTruncate,
  useNewTabIcon
}: AddressLinkProps) => {
  // prioritize robust > id, use id if no robust exists
  const linkText = useMemo(() => {
    if (address) return shouldTruncate ? truncateAddress(address) : address
    return id || ''
  }, [address, id, shouldTruncate])
  const copyText = address || id
  const linkHref = `${explorerUrl}/actor/?address=${copyText}`
  return (
    <LabeledLink
      label={label}
      color={color}
      href={linkHref}
      linkText={linkText}
      copyText={copyText}
      disableLink={disableLink}
      hideCopy={hideCopy}
      hideCopyText={hideCopyText}
      useNewTabIcon={useNewTabIcon}
    />
  )
}

export type AddressLinkProps = {
  id?: string
  address?: string
  shouldTruncate?: boolean
} & Omit<LabeledLinkProps, 'href' | 'linkText' | 'copyText'>

const { href, linkText, copyText, ...addressLinkPropTypes } =
  LabeledLinkPropTypes

AddressLink.propTypes = {
  id: PropTypes.string,
  address: PropTypes.string,
  shouldTruncate: PropTypes.bool,
  ...addressLinkPropTypes
}

AddressLink.defaultProps = {
  shouldTruncate: true
}
