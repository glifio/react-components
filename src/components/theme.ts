export const baseColors = {
  mono: {
    black: '#000',
    nearblack: '#262626',
    darkgray: '#666666',
    lightgray: '#C4C4C4',
    silver: '#999999',
    white: '#ffffff',
    transparent: 'transparent'
  },
  blue: {
    lightest: '#F6F8FE',
    lighter: '#EFF3FD',
    light: '#E4EBFC',
    mid: '#d1ddfa',
    mid2: '#bacbf7',
    muted: '#e8ecf7',
    muted700: '#d9e0f2'
  },
  green: {
    primary: '#1AD08F',
    light: '#D2F5ED',
    muted: '#85e0cb',
    dark: '#007056',
    darker: '#08442F'
  },
  red: {
    light: '#FC6D6F',
    dark: '#660000',
    base: '#ff4242'
  },
  orange: {
    light: '#FF9E80',
    dark: '#661800',
    mid: '#FFA826'
  },
  yellow: {
    light: '#FFDC99',
    deep: '#FCA703',
    mid: '#F7C103'
  },
  purple: {
    light: '#E0D7FE',
    mid: '#5E26FF',
    deep: '#290a85'
  }
}

// The core color object is the only non-semantic object here. This is because these colors are used so widely that it would be highly redundant to replicate these color values repeatedly within this "colors" object to style the text/background of every single Component the app renders.
const core = {
  primary: baseColors.purple.mid,
  secondary: baseColors.purple.light,
  tertiary: baseColors.purple.deep,
  black: baseColors.mono.black,
  nearblack: baseColors.mono.nearblack,
  darkgray: baseColors.mono.darkgray,
  lightgray: baseColors.mono.lightgray,
  silver: baseColors.mono.silver,
  white: baseColors.mono.white,
  transparent: baseColors.mono.transparent,
  yellow: baseColors.yellow.deep
}

const colors = {
  core,
  buttons: {
    primary: {
      background: baseColors.green.primary,
      borderColor: baseColors.green.primary,
      color: baseColors.green.darker
    },
    secondary: {
      background: baseColors.mono.transparent,
      borderColor: baseColors.mono.nearblack,
      color: baseColors.mono.nearblack
    },
    tertiary: {
      background: baseColors.mono.transparent,
      borderColor: baseColors.purple.light,
      color: baseColors.purple.light
    },
    'tertiary-selected': {
      background: baseColors.purple.mid,
      borderColor: baseColors.purple.mid,
      color: baseColors.purple.light
    }
  },
  background: {
    app: baseColors.blue.lightest,
    screen: baseColors.blue.lighter,
    text: baseColors.blue.light,
    messageHistory: baseColors.blue.muted
  },
  card: {
    account: {
      background: core.primary,
      color: core.secondary
    },
    balance: {
      background: baseColors.mono.transparent,
      color: core.nearblack
    },
    confirmation: {
      background: baseColors.green.light,
      foreground: baseColors.green.dark
    },
    error: {
      background: baseColors.red.light,
      foreground: baseColors.red.dark
    }
  },
  input: {
    background: {
      base: baseColors.blue.mid,
      active: baseColors.blue.mid2,
      valid: baseColors.green.muted,
      invalid: baseColors.red.light
    },
    border: core.silver
  },
  status: {
    success: {
      background: baseColors.green.primary,
      foreground: baseColors.green.darker
    },
    pending: {
      background: baseColors.yellow.deep,
      foreground: baseColors.mono.darkgray
    },
    warning: {
      background: baseColors.orange.light,
      foreground: baseColors.orange.dark
    },
    fail: {
      background: baseColors.red.light,
      foreground: baseColors.red.dark
    },
    inactive: baseColors.mono.lightgray
  },
  // gonna start slowly migrating colors from their inheritance vars above
  blue: {
    muted700: '#d9e0f2'
  },
  gray: {
    light: '#C4C4C4',
    medium: '#949598',
    dark: '#666666'
  },
  green: {
    primary: '#1AD08F',
    light: '#D2F5ED',
    muted: '#85e0cb',
    medium: '#007B39',
    dark: '#007056',
    darker: '#08442F'
  },
  yellow: {
    light: '#FFDC99',
    medium: '#C59900',
    deep: '#FCA703',
    mid: '#F7C103'
  },
  purple: {
    light: '#E0D7FE',
    medium: '#5E26FF',
    deep: '#290a85'
  }
}

const theme = {
  colors,
  fontSizes: [
    '0rem',
    '1rem',
    '1.125rem',
    '1.25rem',
    '1.5rem',
    '2rem',
    '3rem',
    '5rem'
  ],
  fontWeights: [0, 400, 600, 900],
  letterSpacings: [0, 1, 2, 4, 8],
  lineHeights: {
    solid: 1,
    title: 1.2,
    copy: 1.4
  },
  textStyles: {
    header: {
      fontSize: 'calc(48px + (48 - 32) * (100vw - 360px) / (1440 - 360))',
      fontWeight: 400,
      margin: 0,
      lineHeight: 'title',
      fontFamily: 'RT-Alias-Grotesk'
    },
    bigTitle: {
      fontSize: 6,
      fontWeight: 700,
      margin: 0,
      lineHeight: 'solid',
      fontFamily: 'RT-Alias-Medium'
    },
    title: {
      fontSize: 4,
      fontWeight: 400,
      margin: 0,
      lineHeight: 'title',
      fontFamily: 'RT-Alias-Grotesk'
    },
    text: {
      fontSize: 2,
      fontColor: core.nearblack,
      fontWeight: 400,
      lineHeight: 'copy',
      fontFamily: 'RT-Alias-Grotesk'
    },
    label: {
      fontSize: 2,
      fontColor: core.darkgray,
      fontWeight: 400,
      lineHeight: 'solid',
      fontFamily: 'RT-Alias-Grotesk'
    },
    num: {
      m: {
        fontSize: 4,
        fontWeight: 400,
        margin: 0,
        lineHeight: 'title',
        fontFamily: 'RT-Alias-Grotesk'
      },
      l: {
        fontSize: 5,
        fontWeight: 700,
        margin: 0,
        lineHeight: 'title',
        fontFamily: 'RT-Alias-Grotesk'
      },
      xl: {
        fontSize: 6,
        fontWeight: 700,
        margin: 0,
        lineHeight: 'solid',
        fontFamily: 'RT-Alias-Grotesk'
      },
      xxl: {
        fontSize: 7,
        fontWeight: 700,
        margin: 0,
        lineHeight: 'solid',
        fontFamily: 'RT-Alias-Grotesk'
      }
    }
  },
  fonts: {
    AliasMedium:
      '"RT-Alias-Medium", "system-ui", "Segoe UI", "Roboto", Helvetica',
    AliasGrotesk:
      '"RT-Alias-Grotesk", "system-ui", "Segoe UI", "Roboto", Helvetica',
    sansSerif:
      '"system-ui", "Segoe UI", Roboto, Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";',
    mono: '"system-ui", "Segoe UI", Roboto Mono, Helvetica, Arial, monospace, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";'
  },
  space: [0, 4, 8, 16, 24, 32, 48, 64, 128, 256],
  sizes: [
    0, 4, 8, 16, 24, 32, 48, 64, 80, 160, 240, 300, 480, 560, 640, 720, 800,
    960, 1024, 1200, 1440
  ],
  radii: ['0', '1px', '4px', '8px', '16px', '32px', '100px'],
  borders: [0, `1px solid`, `4px solid`],
  borderStyles: ['solid'],
  borderWidths: ['0', '1px', '2px', '4px'],
  breakpoints: ['40em', '52em', '64em'],
  shadows: [
    '0',
    '0 0.7px 2.2px -8px rgba(0, 0, 0, 0.18),0 1.7px 5.4px  rgba(0, 0, 0, 0.129),0 3.1px 10.1px rgba(0, 0, 0, 0.107),0 5.6px 18.1px rgba(0, 0, 0, 0.09),0 10.4px 33.8px rgba(0, 0, 0, 0.073),0 25px 81px rgba(0, 0, 0, 0.051)',
    'rgba(0, 0, 0, 0.10) 0px 0.7px 2.2px -8px, rgba(0, 0, 0, 0.04) 0px 1.7px 2.4px, rgba(0, 0, 0, 0.106) 0px 3.1px 8.1px, rgba(0, 0, 0, 0.04) 0px 5.6px 12.1px, rgba(0, 0, 0, 0.045) 0px 4.4px 4.8px, rgba(0, 0, 0, 0.05) 0px 15px 41px'
  ],
  opacity: {
    disabled: 0.4
  },
  zIndices: [0, 9, 99, 999, 9999]
}

export default theme

/* EVERYTHING BELOW HERE IS THEMEV2 */

// safe to change
const deviceWidthPxs = {
  phone: 480,
  tablet: 868,
  desktop: 1124
}

const spaces = {
  phone: {
    sm: '',
    small: '',
    default: '8px',
    md: '8px',
    medium: '8px',
    lg: '20px',
    large: '20px'
  },
  default: {
    sm: '',
    small: '',
    default: '10px',
    md: '10px',
    medium: '10px',
    lg: '20px',
    large: '20px'
  },
  tablet: {
    sm: '',
    small: '',
    default: '10px',
    md: '10px',
    medium: '10px',
    lg: '20px',
    large: '20px'
  },
  desktop: {
    sm: '',
    small: '',
    default: '10px',
    md: '10px',
    medium: '10px',
    lg: '20px',
    large: '20px'
  }
}

const fontSizes = {
  phone: {
    sm: '13px',
    small: '13px',
    default: '16px',
    md: '18px',
    medium: '18px',
    lg: '20px',
    large: '20px'
  },
  default: {
    sm: '13px',
    small: '13px',
    default: '18px',
    md: '26px',
    medium: '26px',
    lg: '28px',
    large: '28px'
  },
  tablet: {
    sm: '13px',
    small: '13px',
    default: '18px',
    md: '26px',
    medium: '26px',
    lg: '28px',
    large: '28px'
  },
  desktop: {
    sm: '13px',
    small: '13px',
    default: '18px',
    md: '26px',
    medium: '26px',
    lg: '28px',
    large: '28px'
  }
}

export const maxWidth = '1450px'

// dont change
export const devices = {
  phone: `${deviceWidthPxs.phone}px`,
  tablet: `${deviceWidthPxs.tablet}px`,
  desktop: `${deviceWidthPxs.desktop}px`
}

// these are
// one of size = sm, default, md, lg
// one of device = phone, tablet, desktop
export function space(size = 'default', device = 'default') {
  return spaces[device][size]
}

export function fontSize(size = 'default', device = 'default') {
  return fontSizes[device][size]
}
