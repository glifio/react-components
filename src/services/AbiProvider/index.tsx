import { createContext, ReactNode, useContext } from 'react'
import { ABI } from '@glif/filecoin-actor-utils'
import { useLocalStorage } from '../../utils'

export interface AbiContextType {
  abis: Record<string, ABI>
  setAbi: (key: string, val: string) => void
  rmAbi: (key: string) => void
}

export const AbiContext = createContext<ABI>(null)

export const AbiProvider = ({ children }: AbiProviderProps) => {
  const [abis, setAbi, rmAbi] = useLocalStorage<ABI>('ABI')
  return (
    <AbiContext.Provider value={{ abis, setAbi, rmAbi }}>
      {children}
    </AbiContext.Provider>
  )
}

export const useAbi = (address: string): ABI | null => {
  const { abis, setAbi, rmAbi } = useContext(AbiContext)
  return [abis?.[address], setAbi, rmAbi]
}

type AbiProviderProps = {
  children: ReactNode
}
