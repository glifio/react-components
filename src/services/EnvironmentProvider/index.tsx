import LotusRpcEngine from '@glif/filecoin-rpc-client'
import { CoinType } from '@glif/filecoin-address'
import { Logger, LogLevel } from '@glif/logger'
import { useRouter } from 'next/router'
import {
  createContext,
  useContext,
  ReactNode,
  useState,
  useMemo,
  useEffect,
  useCallback
} from 'react'
import pick from 'lodash.pick'
import isEqual from 'lodash.isequal'
import { getQueryParam, switchNetworkUrl } from '../../utils/urlParams'

export enum Network {
  MAINNET = 'mainnet',
  CALIBRATION = 'calibrationnet',
  HYPERSPACE = 'hyperspacenet'
}

export type NetworkInfo = Pick<
  EnvironmentContextType,
  | 'nodeStatusApiKey'
  | 'graphUrl'
  | 'graphSecure'
  | 'lotusApiUrl'
  | 'networkName'
  | 'coinType'
>

const networkInfoKeys: Array<keyof NetworkInfo> = [
  'nodeStatusApiKey',
  'graphUrl',
  'graphSecure',
  'lotusApiUrl',
  'networkName',
  'coinType'
]

export interface EnvironmentContextType {
  homeUrl: string
  walletUrl: string
  safeUrl: string
  explorerUrl: string
  verifierUrl: string
  nodesUrl: string
  blogUrl: string
  githubUrl: string
  discordUrl: string
  twitterUrl: string
  contactEmail: string
  nodeStatusApiUrl: string
  nodeStatusApiKey: string
  graphUrl: string
  graphSecure: boolean
  lotusApiUrl: string
  lotusApi: LotusRpcEngine
  coinType: CoinType
  isProd: boolean
  networkName: Network
  setNetwork: (network: NetworkInfo) => void
  sentryDsn?: string
  sentryEnv?: string
  packageName?: string
  packageVersion?: string
}

export const emptyEnvironmentContext = {
  homeUrl: 'https://apps.glif.io',
  walletUrl: 'https://wallet.glif.io',
  safeUrl: 'https://safe.glif.io',
  explorerUrl: 'https://explorer.glif.io',
  verifierUrl: 'https://verify.glif.io',
  nodesUrl: 'https://api.node.glif.io',
  blogUrl: 'https://blog.glif.io',
  githubUrl: 'https://github.com/glifio',
  discordUrl: 'https://discord.gg/B9ju5Eu4Rq',
  twitterUrl: 'https://twitter.com/glifio',
  contactEmail: 'squad@infinitescroll.org',
  nodeStatusApiUrl: 'https://api.uptimerobot.com/v2/getMonitors',
  nodeStatusApiKey: '',
  graphUrl: '',
  graphSecure: true,
  lotusApiUrl: '',
  lotusApi: null,
  networkName: Network.CALIBRATION,
  setNetwork: () => {},
  coinType: CoinType.TEST,
  isProd: false,
  sentryDsn: '',
  sentryEnv: ''
}

export const EnvironmentContext = createContext<EnvironmentContextType>(
  emptyEnvironmentContext
)

export const networks: Record<Network, NetworkInfo> = {
  [Network.MAINNET]: {
    nodeStatusApiKey: 'm786191525-b3192b91db66217a44f7d4be',
    graphUrl: 'graph.glif.link/query',
    graphSecure: true,
    lotusApiUrl: 'https://mainnet.glif.host',
    networkName: Network.MAINNET,
    coinType: CoinType.MAIN
  },
  [Network.CALIBRATION]: {
    nodeStatusApiKey: 'm787669344-2a9b90eb03dbff3e503c93c7',
    graphUrl: 'graph-calibration.glif.link/query',
    graphSecure: true,
    lotusApiUrl: 'https://api.calibration.node.glif.io/',
    networkName: Network.CALIBRATION,
    coinType: CoinType.TEST
  },
  [Network.HYPERSPACE]: {
    nodeStatusApiKey: 'm793485228-aeeb03f93349e7fc1596791c',
    graphUrl: 'graph-wallaby.glif.link/query',
    graphSecure: true,
    lotusApiUrl: 'https://api.hyperspace.node.glif.io/rpc/v0',
    networkName: Network.HYPERSPACE,
    coinType: CoinType.TEST
  }
}

export const Environment = ({ children, ...environment }: EnvironmentProps) => {
  return (
    <EnvironmentContext.Provider
      value={{
        ...emptyEnvironmentContext,
        ...environment
      }}
    >
      {children}
    </EnvironmentContext.Provider>
  )
}

// This wrapper is used by the apps, which decorates the Environment from the URL bar
export const EnvironmentProvider = ({
  children,
  ...initialEnvironment
}: Omit<EnvironmentProps, keyof NetworkInfo>) => {
  const router = useRouter()
  const networkParam = getQueryParam.string(router, 'network') as Network
  const network =
    Object.keys(networks).find(
      name => name === networkParam || name === `${networkParam}net`
    ) || Network.MAINNET
  const lotusApi = networks[network].lotusApiUrl
    ? new LotusRpcEngine({
        apiAddress: networks[network].lotusApiUrl
      })
    : null
  const env: Omit<EnvironmentProps, 'children'> = {
    ...initialEnvironment,
    ...networks[network],
    lotusApi
  }

  const [environment, setEnvironment] = useState({ ...env })
  const setNetwork = useCallback(
    async (n: NetworkInfo) => {
      const url = switchNetworkUrl(router.asPath, n.networkName)
      window?.location?.assign(url)
    },
    [router]
  )

  // updates state based on the url bar changing
  useEffect(() => {
    const shouldChangeNetwork = !isEqual(
      pick(environment, networkInfoKeys),
      networks[network]
    )
    if (shouldChangeNetwork) {
      const newNetwork = networks[network]
      setEnvironment({
        ...environment,
        ...newNetwork,
        lotusApi: newNetwork.lotusApiUrl
          ? new LotusRpcEngine({
              apiAddress: newNetwork.lotusApiUrl
            })
          : null
      })
    }
  }, [network, environment, setNetwork])

  return (
    <Environment {...env} setNetwork={setNetwork}>
      {children}
    </Environment>
  )
}

export const useEnvironment = (): EnvironmentContextType => {
  return useContext(EnvironmentContext)
}

export const useLogger = (): Logger => {
  const { isProd, sentryDsn, sentryEnv, packageName, packageVersion } =
    useEnvironment()
  return useMemo(
    () =>
      new Logger({
        consoleEnabled: true,
        consoleLevel: LogLevel.DEBUG,
        sentryEnabled: isProd,
        sentryLevel: LogLevel.WARN,
        sentryDsn: sentryDsn || '',
        sentryEnv: sentryEnv || '',
        sentryTraces: 0,
        packageName: packageName || 'react-components',
        packageVersion: packageVersion || '?.?.?'
      }),
    [isProd, sentryDsn, sentryEnv, packageName, packageVersion]
  )
}

export type EnvironmentProps = {
  children: ReactNode
} & Partial<EnvironmentContextType>
