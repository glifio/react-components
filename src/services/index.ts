export * from './AbiProvider'
export * from './WalletProvider/useWallet'
export * from './WalletProvider'
export * from './WalletProvider/state'
export { default as WalletProviderWrapper } from './WalletProvider'
export * from './WalletProvider/types'
export * from './UpdateBalance'
export * from './RequireWallet'
export * from './EnvironmentProvider'
