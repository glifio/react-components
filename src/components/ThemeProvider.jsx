import PropTypes from 'prop-types'
import { ThemeProvider as StyledThemeProvider } from 'styled-components'
import { theme as defaultTheme } from './theme'

const ThemeProvider = ({ children, theme, ...rest }) => {
  return (
    <StyledThemeProvider theme={theme} {...rest}>
      {children}
    </StyledThemeProvider>
  )
}

ThemeProvider.propTypes = {
  theme: PropTypes.object,
  children: PropTypes.node
}

ThemeProvider.defaultProps = {
  theme: defaultTheme,
  children: <></>
}

ThemeProvider.displayName = 'ThemeProvider'

export default ThemeProvider
