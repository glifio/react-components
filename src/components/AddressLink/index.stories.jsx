import theme from '../theme'
import ThemeProvider from '../ThemeProvider'
import { AddressLink } from './index'

export default {
  title: 'AddressLink/AddressLink',
  component: AddressLink,
  decorators: [
    Story => (
      <div style={{ display: 'flex', justifyContent: 'center' }}>
        <ThemeProvider theme={theme}>{Story()}</ThemeProvider>
      </div>
    )
  ],
  parameters: { actions: { argTypesRegex: '^on.*' } }
}

const Template = args => <AddressLink {...args} />

export const Labelled = Template.bind({})
Labelled.args = {
  label: 'Safe Address',
  address: 'f17uoq6tp427uzv7fztkbsnn64iwotfrristwpryy'
}

export const Black = Template.bind({})
Black.args = {
  address: 'f17uoq6tp427uzv7fztkbsnn64iwotfrristwpryy',
  color: 'var(--black)'
}

export const AddressOnly = Template.bind({})
AddressOnly.args = {
  address: 'f17uoq6tp427uzv7fztkbsnn64iwotfrristwpryy'
}

export const IDOnly = Template.bind({})
IDOnly.args = {
  id: 't07633'
}

export const AddressAndID = Template.bind({})
AddressAndID.args = {
  id: 't07633',
  address: 'f17uoq6tp427uzv7fztkbsnn64iwotfrristwpryy'
}
