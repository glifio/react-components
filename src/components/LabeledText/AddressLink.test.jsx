import { render } from '@testing-library/react'
import { AddressLink } from './AddressLink'
import theme from '../theme'
import ThemeProvider from '../ThemeProvider'
import { Environment } from '../../services/EnvironmentProvider'

describe('Receive', () => {
  test('it renders correctly', () => {
    const { container } = render(
      <Environment>
        <ThemeProvider theme={theme}>
          <AddressLink
            label='Safe Address'
            address='t137sjdbgunloi7couiy4l5nc7pd6k2jmq32vizpy'
          />
        </ThemeProvider>
      </Environment>
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
