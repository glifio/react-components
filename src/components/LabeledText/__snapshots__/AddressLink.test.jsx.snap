// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Receive it renders correctly 1`] = `
.c0 h4 {
  margin: 0;
  color: var(--gray-dark);
}

.c0 p {
  margin: 0;
  line-height: 1.5;
}

.c2 {
  min-width: 0;
  box-sizing: border-box;
  color: var(--purple-medium);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  width: 24;
  height: 24;
}

.c4 {
  height: auto;
  cursor: pointer;
  opacity: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: 0;
  background: transparent;
  padding: 0;
  outline: none;
}

.c7 {
  -webkit-transition: 0.24s ease-in-out;
  transition: 0.24s ease-in-out;
}

.c3:hover .c6 {
  -webkit-transform: scale(1.25);
  -ms-transform: scale(1.25);
  transform: scale(1.25);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  grid-gap: 0.25em;
  line-height: 1.5;
}

.c1 > a {
  color: var(--purple-medium);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1 > a:hover {
  color: var(--purple-medium);
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c1 > a svg {
  -webkit-transition: 0.24s ease-in-out;
  transition: 0.24s ease-in-out;
  vertical-align: middle;
}

.c1 > a svg:hover {
  -webkit-transform: scale(1.25);
  -ms-transform: scale(1.25);
  transform: scale(1.25);
  fill: var(--purple-medium);
}

<div
  class="c0"
>
  <h4>
    Safe Address
  </h4>
  <div
    class="c1"
    color="var(--purple-medium)"
  >
    <a
      href="https://explorer.glif.io/actor/?address=t137sjdbgunloi7couiy4l5nc7pd6k2jmq32vizpy"
      rel="noreferrer noopener"
      target="_blank"
    >
      t137sj ... 2vizpy
    </a>
    <div
      class="c2"
      color="var(--purple-medium)"
      display="flex"
    >
      <button
        class="c3 c4"
        display="flex"
        role="button"
        type="button"
      >
        <svg
          class="c5 c6 c7"
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="#1A0066"
            height="12"
            width="12"
            x="8"
            y="4"
          />
          <rect
            fill="#E0D7FE"
            height="12"
            width="12"
            x="4"
            y="8"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
